File: app/src/main/java/com/veganbeauty/admin/data/local/dao/AdminDao.java
package com.veganbeauty.admin.data.local.dao;

import androidx.annotation.Nullable;
import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import com.veganbeauty.admin.data.local.entities.AdminEntity;

import java.util.List;

@Dao
public interface AdminDao {

    @Query("SELECT * FROM admins")
    LiveData<List<AdminEntity>> getAllLiveData();

    @Query("SELECT * FROM admins")
    List<AdminEntity> getAllSync();

    @Query("SELECT * FROM admins WHERE username = :username")
    @Nullable
    AdminEntity getByUsernameSync(String username);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertSync(AdminEntity admin);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAllSync(List<AdminEntity> admins);

    @Query("DELETE FROM admins")
    void deleteAllSync();
}

File: app/src/main/java/com/veganbeauty/admin/data/local/dao/AdminDao.kt
package com.veganbeauty.admin.data.local.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import com.veganbeauty.admin.data.local.entities.AdminEntity

@Dao
interface AdminDao {

    @Query("SELECT * FROM admins")
    fun getAllLiveData(): LiveData<List<AdminEntity>>

    @Query("SELECT * FROM admins")
    fun getAllSync(): List<AdminEntity>

    @Query("SELECT * FROM admins WHERE username = :username")
    fun getByUsernameSync(username: String): AdminEntity?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertSync(admin: AdminEntity)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAllSync(admins: List<AdminEntity>)

    @Query("DELETE FROM admins")
    fun deleteAllSync()
}
